pages/[category]/[artwork].vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <div class="absolute z-10 bg-white h-screen w-screen overflow-hidden border-blue" :class="transition">
+    <MoleculesHeader :showCategory="true">
+      <span class="text-3xl uppercase">{{ currentCategory?.title }} {{ currentCategory?.period }}</span>
+    </MoleculesHeader>
+    <div class="grid grid-cols-12">
+      <div class="col-span-1 flex justify-center items-center">
+        <MoleculesIconNavigation name="arrowLeft" :to="navPath(true)" direction="right" v-if="state.prev"/>
+      </div>
+      <div class="col-span-10 flex flex-col justify-center items-center">
+        <div class="placeholder w-full bg-blue flex justify-center items-center">
+          <img :src="currentArtwork?.image" v-if="currentArtwork?.image" class="h-full object-cover" />
```

Comment:
Artworks kennen volgens de type geen optionele properties.
Hier kan het natuurlijk zijn dat:
- Artwork nog geladen moet worden.
- Artwork niet gevonden kon worden.
- Artwork om andere reden niet beschikbaar is.

Dat eerste kan echter zover ik de logica volg niet voorkomen in server/client, omdat er middleware is die redirect als er nog geen data is én client side data prefetched wordt.
Het tweede kan ook niet voorkomen om diezelfde reden.

Dus is er altijd een currentArtwork aanwezig zodra deze code wordt uitgevoerd.

Reply:
Ik ben het helemaal met je eens, daarentegen als ik het run, kreeg ik regelmatig title of undefined meldingen op de template van een artwork. Ik kan gebruik maken van computed properties om het op te lossen maar dat verplaatst het probleem alleen, dan gebruik ik liever deze methode, waardoor de default waarde terugvalt op undefined en dus niks rendert.

Reply:
Dat betekent dat mijn redenering niet klopt. Ben wel benieuwd hoe dat zit.

Een alternatieve oplossing is om hogerop een conditional op te nemen zodat je iets anders rendert als er (nog) geen `currentArtwork` is. Dan render je bijvoorbeeld een skeleton of een loading indicator indien een load in gang is.
Als er geen artwork gevonden is kun je een 404 gooien: https://v3.nuxtjs.org/getting-started/error-handling#createerror